Workbooks/Azure Update Management/Azure Update Management.workbook

```diff
@@ -0,0 +1,668 @@
ate\r\n    | where TimeGenerated>ago(14h) and OSType!=\"Linux\"\r\n    | summarize hint.strategy=partitioned arg_max(TimeGenerated, UpdateState, Approved, Optional, Classification) by SourceComputerId, UpdateID\r\n    | distinct SourceComputerId, Classification, UpdateState, Approved, Optional\r\n    | summarize WorstMissingUpdateSeverity=max(iff(UpdateState=~\"Needed\" and (Optional==false or Classification has \"Critical\" or Classification has \"Security\") and Approved!=false, iff(Classification has \"Critical\", 4, iff(Classification has \"Security\", 2, 1)), 0)) by SourceComputerId\r\n)\r\non SourceComputerId\r\n| extend WorstMissingUpdateSeverity=coalesce(WorstMissingUpdateSeverity, -1)\r\n| summarize computersBySeverity=count() by WorstMissingUpdateSeverity\r\n| union (Heartbeat\r\n| where TimeGenerated>ago(12h) and OSType==\"Linux\" and notempty(Computer)\r\n| summarize arg_max(TimeGenerated, Solutions) by SourceComputerId\r\n| where Solutions has \"updates\"\r\n| distinct SourceComputerId\r\n| join kind=leftouter\r\n(\r\n    Update\r\n    | where TimeGenerated>ago(5h) and OSType==\"Linux\"\r\n    | summarize hint.strategy=partitioned arg_max(TimeGenerated, UpdateState, Classification) by SourceComputerId, Product, ProductArch\r\n    | distinct SourceComputerId, Classification, UpdateState\r\n    | summarize WorstMissingUpdateSeverity=max(iff(UpdateState=~\"Needed\", iff(Classification has \"Critical\", 4, iff(Classification has \"Security\", 2, 1)), 0)) by SourceComputerId\r\n)\r\non SourceComputerId\r\n| extend WorstMissingUpdateSeverity=coalesce(WorstMissingUpdateSeverity, -1)\r\n| summarize computersBySeverity=count() by WorstMissingUpdateSeverity)\r\n| summarize assessedComputersCount=sumif(computersBySeverity, WorstMissingUpdateSeverity>-1), notAssessedComputersCount=sumif(computersBySeverity, WorstMissingUpdateSeverity==-1), computersNeedCriticalUpdatesCount=sumif(computersBySeverity, WorstMissingUpdateSeverity==4), computersNeedSecurityUpdatesCount=sumif(computersBySeverity, WorstMissingUpdateSeverity==2), computersNeedOtherUpdatesCount=sumif(computersBySeverity, WorstMissingUpdateSeverity==1), upToDateComputersCount=sumif(computersBySeverity, WorstMissingUpdateSeverity==0)\r\n| summarize assessedComputersCount=sum(assessedComputersCount), computersNeedCriticalUpdatesCount=sum(computersNeedCriticalUpdatesCount),  computersNeedSecurityUpdatesCount=sum(computersNeedSecurityUpdatesCount), computersNeedOtherUpdatesCount=sum(computersNeedOtherUpdatesCount), upToDateComputersCount=sum(upToDateComputersCount), notAssessedComputersCount=sum(notAssessedComputersCount)\r\n| extend allComputersCount=assessedComputersCount+notAssessedComputersCount\r\n| project machinePack = pack(\"Assessed machines\", assessedComputersCount, \"Not assessed machines\", notAssessedComputersCount)\r\n| mv-expand machinePack\r\n| extend machine = tostring(bag_keys(machinePack)[0])\r\n| extend count_ = tolong(machinePack[machine])\r\n| project machine, count_ ",
```

Comment:
Same here, I guess general question for the whole workbook: Is timerange parameter needed?